### ISSUE TITLE
[solution] Virtualbox 5.2 not compatible with Vagrant 1.9.1

### ISSUE THREAD

**Issue by nhegde610** [rapid7/metasploitable3#220]
## Issue Description
The ReadMe File specifies the use of vagrant 1.9.1 however once the build is finished and the command "vagrant up" is issued, vagrant does not recognize virtuabox-5.2 as a default provider. It throws error stating no default provider on the system.

## Host System
* OS: Kali-rolling (latest-version)
* Packer Version: 1.1.3
* Vagrant Version: 1.9.1
* VirtualBox Version: 5.2

## Command Output
The output of the successful build using vagrant 1.9.1

> ==> Builds finished. The artifacts of successful builds are:
--> virtualbox-iso: 'virtualbox' provider box: windows_2008_r2_virtualbox.box
Box successfully built by Packer.
Attempting to add the box to Vagrant...
==> box: Box file was not detected as metadata. Adding it directly...
==> box: Adding box 'metasploitable3' (v0) for provider: 
    box: Unpacking necessary files from: file:///root/metasploitable3-master/windows_2008_r2_virtualbox.box
==> box: Successfully added box 'metasploitable3' (v0) for 'virtualbox'!
Box successfully added to Vagrant.
---------------------------------------------------------------------

> SUCCESS: Run 'vagrant up' to provision and start metasploitable3.
NOTE: The VM will need Internet access to provision properly.

However, while running vagrant 1.9.1 gives this:
root@kali:~/metasploitable3-master# vagrant up
>No usable default provider could be found for your system.

>Vagrant relies on interactions with 3rd party systems, known as
"providers", to provide Vagrant with resources to run development
environments. Examples are VirtualBox, VMware, Hyper-V.

>The easiest solution to this message is to install VirtualBox, which
is available for free on all major platforms.

>If you believe you already have a provider available, make sure it
is properly installed and configured. You can see more details about
why a particular provider isn't working by forcing usage with
`vagrant up --provider=PROVIDER`, which should give you a more specific
error message for that particular provider.

Running the command with provider flag does no good. So, I uninstalled vagrant 1.9.1 using `sudo apt-get remove vagrant` . Then, I installed the latest vagrant 2.0.1 and updated the plugin using the command : `vagrant plugin update` 

and ran the command vagrant up : 
root@kali:~/metasploitable3-master# vagrant up
>Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'metasploitable3'...
==> default: Matching MAC address for NAT networking...
==> default: Setting the name of the VM: metasploitable3-master_default_1517075576398_66066
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 3389 (guest) => 3389 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
    default: 5985 (guest) => 55985 (host) (adapter 1)
    default: 5986 (guest) => 55986 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: WinRM address: 127.0.0.1:55985
    default: WinRM username: vagrant
    default: WinRM execution_time_limit: PT2H
    default: WinRM transport: negotiate
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
==> default: Mounting shared folders...
    default: /vagrant => /root/metasploitable3-master
==> default: Running provisioner: shell...
    default: Running: scripts/configs/enable_firewall.bat as c:\tmp\vagrant-shell.bat
    default: Ok.
==> default: Running provisioner: shell...
    default: Running: scripts/configs/configure_firewall.bat as c:\tmp\vagrant-shell.bat
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
    default: Ok.
   root@kali:~/metasploitable3-master# 

and it turns out that I am able to access the virtual machine from the virtualbox 

![screenshot from 2018-01-27 23-56-11](https://user-images.githubusercontent.com/18752973/35475056-b472fe16-03bd-11e8-8d08-ed39f6b6d016.png)

**Comment by PeterisP** [comment]
Vagrant 1.9.1 works with VirtualBox 5.2 with the following simple modification (from https://github.com/hashicorp/vagrant/issues/9090#issuecomment-339935445):

Add "5.2" => Version_5_1, to /opt/vagrant/embedded/gems/gems/vagrant-1.9.1/plugins/providers/virtualbox/driver/meta.rb  after row 64

However, if Vagrant 1.9.1 is no longer a requirement and the latest Vagrant can be used, that's probably a better way to go.

**Comment by robertodijo** [comment]
> Vagrant 1.9.1 works with VirtualBox 5.2 with the following simple modification (from [hashicorp/vagrant#9090 (comment)](https://github.com/hashicorp/vagrant/issues/9090#issuecomment-339935445)):
> 
> Add "5.2" => Version_5_1, to /opt/vagrant/embedded/gems/gems/vagrant-1.9.1/plugins/providers/virtualbox/driver/meta.rb after row 64
> 
> However, if Vagrant 1.9.1 is no longer a requirement and the latest Vagrant can be used, that's probably a better way to go.

Thanks, you save me a lot of debugging time. This works.

**Comment by HenryPineda** [comment]
@PeterisP solution worked for me. Thanks!

**Comment by terceranexus6** [comment]
Did you guys encountered any problem using Vagrant 1.9.1 with VirtualBox 6.0.4? Because having both correctly installen in debian jessie, it says "No usable default provider could be found for your system." and it's weird.

**Comment by mkienow-r7** [comment]
@terceranexus6 This issue is closed. Vagrant 1.9.1 was the source of the issue with newer VirtualBox 
 versions, and it is advised to use a newer version of Vagrant. Please try using the latest Vagrant and open a new issue if that doesn't resolve the issue.
